pkg/lso/localvolumediscovery_test.go

```diff
@@ -0,0 +1,391 @@
+	}{
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            nil,
+		},
+		{
+			testLocalVolumeDiscovery: buildInValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            fmt.Errorf("localVolumeDiscovery 'name' cannot be empty"),
+		},
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError: fmt.Errorf("localvolumediscoveries.local.storage.openshift.io " +
+				"\"auto-discover-devices\" not found"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		localVolumeDiscoveryObj, err := testCase.testLocalVolumeDiscovery.Get()
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, localVolumeDiscoveryObj.Name, testCase.testLocalVolumeDiscovery.Definition.Name)
+			assert.Equal(t, localVolumeDiscoveryObj.Namespace, testCase.testLocalVolumeDiscovery.Definition.Namespace)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestLocalVolumeDiscoveryCreate(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeDiscovery *LocalVolumeDiscoveryBuilder
+		expectedError            error
+	}{
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            nil,
+		},
+		{
+			testLocalVolumeDiscovery: buildInValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            fmt.Errorf("localVolumeDiscovery 'name' cannot be empty"),
+		},
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:            nil,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testLocalVolumeSetBuilder, err := testCase.testLocalVolumeDiscovery.Create()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, testLocalVolumeSetBuilder.Definition.Name, testLocalVolumeSetBuilder.Object.Name)
+			assert.Equal(t, testLocalVolumeSetBuilder.Definition.Namespace, testLocalVolumeSetBuilder.Object.Namespace)
+			assert.Nil(t, err)
+		}
+	}
+}
+
+func TestLocalVolumeDiscoveryDelete(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeDiscovery *LocalVolumeDiscoveryBuilder
+		expectedError            error
+	}{
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            nil,
+		},
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:            fmt.Errorf("localVolumeDiscovery cannot be deleted because it does not exist"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		err := testCase.testLocalVolumeDiscovery.Delete()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testLocalVolumeDiscovery.Object)
+		}
+	}
+}
+
```

Comment:
Please add test case for IsDiscovering function

Reply:
it's a dummy object, how can I really test discovering state on not real object, it's a status, not spec.
+ it should be other object types involed to test it is really discovering

Reply:
such test is possible only in case I will replace "return builder.Object.Status.Phase, nil" to the "return builder.Definition.Status.Phase, nil" (Object to the Definition). But I am not sure, that in this case this method will work with the real object